include/quic/endpoint.hpp

```diff
@@ -90,12 +90,9 @@ namespace oxen::quic
             std::promise<std::shared_ptr<Connection>> p;
             auto f = p.get_future();
 
-            Path _path;
+            auto& _r = (_local.is_ipv6()) ? Address::map_ipv4_as_ipv6(remote) : remote;
```

Comment:
What happens if `remote` is also/already ipv6?

Reply:
This is making a reference to a temporary.

It would be cleaner to have the mapping function as a mutator of an Address rather than a static method making a new copy of it (because it seems unlikely that where this would be used would ever care about keeping the original).

Reply:
> What happens if `remote` is also/already ipv6?

Then you'll get an error as soon as you try to send anything.  I suppose that error could be detected earlier and throw, though.

Reply:
Yeah I need to account for that error

@jagerman I made it into a static method because the Address passed to connect is a const reference, so I would have had to const cast it

Reply:
fixed and added a test case